tests/datetime/test_dates.py

```diff
@@ -38,9 +41,25 @@ def test_can_find_each_month():
         minimal(dates(), lambda x: x.month == i)
 
 
+@checks_deprecated_behaviour
 def test_min_year_is_respected():
     assert minimal(dates(min_year=2003)).year == 2003
 
 
+@checks_deprecated_behaviour
 def test_max_year_is_respected():
     assert minimal(dates(max_year=1998)).year == 1998
+
+
+def test_min_date_is_respected():
+    d = date(2003, 7, 12)
+    assert minimal(dates(min_date=d)) == d
+
+
+def test_max_date_is_respected():
+    d = date(1998, 5, 17)
+    assert minimal(dates(max_date=d)) == date.min.replace(year=d.year)
```

Comment:
Rather than `date.min.replace()`, can we be explicit? I think it should be `date(1998, 1, 1)`, unless I’ve missed something.